src/components/Header/Header.tsx

```diff
@@ -1,53 +1,113 @@
-import Button from '../Button/Button';
+import classnames from 'classnames';
+import * as React from 'react';
+import { px } from '../../utils';
+// import Navigation from '../Navigation/Navigation';
+// import NavigationItem from '../Navigation/NavigationItem/NavigationItem';
+// import NavigationItemTrigger from '../Navigation/NavigationItemTrigger/NavigationItemTrigger';
+// import NavigationList from '../Navigation/NavigationList/NavigationList';
+import Search from '../Search/Search';
+// import { LinkVariants } from '../Link/utils';
 
-// Try to not pass complex objects into components
-type User = {
-  name: string;
+export interface HeaderProps extends React.HTMLAttributes<HTMLElement> {
+  /**
+   * Logo src
+   */
+  logo?: string;
+  /**
+   * Logo alt text
+   */
+  logoText?: string;
+  /**
+   * Toggle open text
+   */
+  toggleOpenText?: string;
+  /**
+   * Toggle close text
+   */
+  toggleCloseText?: string;
+}
+type HeaderContextType = {
+  expandedItem: string;
+  setExpandedItem: (item: string) => void;
+  expanded: boolean;
+  handleSelection: (label: string) => void;
+  handleFocusEvent: (e: React.FocusEvent) => void;
 };
 
-export interface HeaderProps {
-  user?: User;
-  onLogin?: () => void;
-  onLogout?: () => void;
-  onCreateAccount?: () => void;
-}
+const defaultMobileMenuLabel = 'Main Menu';
+export const HeaderContext = React.createContext({
+  expandedItem: defaultMobileMenuLabel,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  setExpandedItem: (_item: string) => {},
+  expanded: false,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  handleSelection: (_label: string) => {},
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  handleFocusEvent: (_e: React.FocusEvent) => {},
+});
 
-const Header = ({ user, onLogin, onLogout, onCreateAccount }: HeaderProps) => (
-  <header>
-    <div className="storybook-header">
-      <div>
-        <svg width="32" height="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
-          <g fill="none" fillRule="evenodd">
-            <path d="M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z" fill="#FFF" />
-            <path d="M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z" fill="#555AB9" />
-            <path d="M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z" fill="#91BAF8" />
-          </g>
-        </svg>
-        <h1>Acme</h1>
+const Header = ({
+  logo,
+  logoText,
+  className,
+  children,
+  toggleOpenText = 'Open Menu',
+  toggleCloseText = 'Close Menu',
+}: React.PropsWithChildren<HeaderProps>) => {
+  const [toggleText, setToggleText] = React.useState(toggleOpenText);
```

Comment:
the `toggleText` can be derived from the `toggleState` so we don't need a separate state here.